src/custom/state/claim/hooks/index.ts

```diff
@@ -442,75 +444,129 @@ export function useClaimCallback(account: string | null | undefined): {
-        !connectedAccount ||
-        !chainId ||
-        !vCowContract ||
-        !vCowToken ||
-        !nativeTokenPrice
-      ) {
-        throw new Error("Not initialized, can't claim")
+  const getClaimArgs = useCallback(
+    async function (claimInput: ClaimInput[]): Promise<GetClaimManyArgsResult> {
+      if (claims.length === 0) {
+        throw new Error('User has no claims')
+      }
+      if (claimInput.length === 0) {
+        throw new Error('No claims selected')
+      }
+      if (!account) {
+        throw new Error('Claim account not set')
+      }
+      if (!connectedAccount) {
+        throw new Error('Not connected')
+      }
+      if (!nativeTokenPrice) {
+        throw new Error("There's no native token price")
       }
 
       _validateClaimable(claims, claimInput, isInvestmentWindowOpen, isAirdropWindowOpen)
 
-      const { args, totalClaimedAmount } = _getClaimManyArgs({
+      return _getClaimManyArgs({
         claimInput,
         claims,
         account,
         connectedAccount,
         nativeTokenPrice,
       })
+    },
+    [account, claims, connectedAccount, isAirdropWindowOpen, isInvestmentWindowOpen, nativeTokenPrice]
+  )
+
+  const estimateGasCallback = useCallback(
+    async function (
+      claimInput: ClaimInput[],
+      claimArgs?: GetClaimManyArgsResult['args']
+    ): Promise<BigNumber | undefined> {
+      if (!vCowContract) {
+        return
+      }
+
+      try {
+        let args = claimArgs
+        if (!claimArgs) {
+          const { args: _args } = await getClaimArgs(claimInput)
+          args = _args
+        }
+
+        if (!args) {
+          console.debug('Failed to estimate gas for claiming: There were no valid claims selected')
+          return
+        }
+
+        // Why unnecessarily awaiting here?
+        // Because I want to handle errors here.
+        // Not awaiting means the caller will have to deal with that, which I don't want in this case
+        return await vCowContract.estimateGas.claimMany(...args)
+      } catch (e) {
+        console.debug('Failed to estimate gas for claiming:', e.message)
+        return
+      }
+    },
+    [getClaimArgs, vCowContract]
+  )
+
+  const claimCallback = useCallback(
+    /**
+     * Claim callback that sends tx to wallet to claim whatever user selected
+     *
+     * Returns a string with the formatted vCow amount being claimed
+     */
+    async function (claimInput: ClaimInput[]): Promise<string> {
+      if (claimInput.length === 0) {
+        throw new Error('No claims selected')
+      }
+      if (!account) {
+        throw new Error('Claim account not set')
+      }
+      if (!connectedAccount) {
+        throw new Error('Not connected')
+      }
+      if (!chainId) {
+        throw new Error('No chainId')
+      }
+      if (!vCowContract) {
+        throw new Error('vCOW contract not present')
+      }
+      if (!vCowToken) {
+        throw new Error('vCOW token not present')
+      }
```

Comment:
Aren't this checks replicated in `getClaimArgs`?

Reply:
Yes but if not checked TS complains